neuralmonkey/learning_utils.py

```diff
@@ -23,16 +23,16 @@
 
 
 # pylint: disable=too-many-arguments, too-many-locals, too-many-branches
-# pylint: disable=too-many-statements
+# pylint: disable=too-many-statements, too-many-nested-blocks
 def training_loop(tf_manager: TensorFlowManager,
                   epochs: int,
                   trainer: GenericTrainer,  # TODO better annotate
                   batch_size: int,
                   train_dataset: Dataset,
-                  val_dataset: Dataset,
                   log_directory: str,
                   evaluators: EvalConfiguration,
                   runners: List[BaseRunner],
+                  val_dataset: Union[Dataset, List[Dataset]],
```

Comment:
asi rozumim, žes to hodil k test_datasets, ale přihodil bych k tomu ještě ten train_dataset, ať jsou datasety pohromadě..